lightning_transformers/task/nlp/question_answering/model.py

```diff
@@ -43,3 +46,29 @@ def training_step(self, batch: Any, batch_idx: int) -> torch.Tensor:
     @property
     def hf_pipeline_task(self) -> str:
         return "question-answering"
+
+    def validation_step(self, batch: Any, batch_idx: int, dataloader_idx: int = 0) -> Dict[str, torch.Tensor]:
+        batch.pop("offset_mapping")
+        example_ids = batch.pop("example_id")
+        outputs = self.model(**batch)
+        loss = outputs[0]
+        self.log("val_loss", loss, prog_bar=True, sync_dist=True)
+        self.metric(example_ids, outputs.start_logits, outputs.end_logits)
+        return loss
+
+    def validation_epoch_end(self, outputs: Any) -> None:
```

Comment:
The metric is only computed over the entire dataset. I wasn't sure where to put that, but this seemed right.

Reply:
Torchmetrics automatically get called `update` on every step and `compute` at the end through the logger, so this isn't needed I think! I will test

Reply:
It looked like `compute` was only being called through `forward`. I initialize the `Metric` with `compute_on_step=False` (the code is not equipped to compute the metric over a partial dataset, but you might be able to modify it to permit that), which makes `forward` only call `update`, not `compute`. I'm not sure if `compute` gets explicitly called anywhere. So yes, please test.

Reply:
Asked someone on the torchmetrics team, your implementation is correct since it's a dict output